runtime/lua/vscode/api.lua

```diff
@@ -418,4 +418,94 @@ do
   end
 end
 
+-----------------------
+------ utilities ------
+-----------------------
+
+---Perform operations in insert mode or switch to insert mode.
+---If in visual mode, this function will preserve the selection after
+---switching to insert mode.
+---
+---Example:
+---Make `editor.action.addSelectionToNextFindMatch` work correctly in any mode.
+---This is the behavior of the default VSCode shortcut Ctrl+d: Add Selection To Next Find Match.
+---
+---```lua
+---vim.keymap.set({ "n", "x", "i" }, "<C-d>", function()
+---  vscode.into_insert(function()
+---    vscode.action("editor.action.addSelectionToNextFindMatch")
+---  end)
+---end)
+---````
+---
+---@param callback function|nil Optional callback function to run after switching to insert mode.
+---                             If not provided, it will directly switch to insert mode synchronously.
+function M.into_insert(callback)
+  vim.validate({ callback = { callback, "f", true } })
+
+  local mode = api.nvim_get_mode().mode
+
+  --- Insert ---
+
+  if mode == "i" then
+    if callback then
+      callback()
+    end
+    return
+  end
+
+  --- Normal ---
+
+  if mode == "n" then
+    api.nvim_feedkeys("a", "n", false)
```

Comment:
Should this be `i` instead of `a`? Which one causes no cursor update (I forget)?

Reply:
You should use "a", you can check the first test